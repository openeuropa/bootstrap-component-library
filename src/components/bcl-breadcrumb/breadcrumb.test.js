import {
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from "@openeuropa/bcl-test-utils";
import { axe, toHaveNoViolations } from "jest-axe";
import demoData from "@openeuropa/bcl-data-breadcrumb/data";

expect.extend(toHaveNoViolations);

const template = "@oe-bcl/bcl-breadcrumb/breadcrumb.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Breadcrumb", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test(`passes the accessibility tests`, async () => {
    expect(
      await axe(renderTwigFileAsHtml(template, demoData))
    ).toHaveNoViolations();
  });
});
