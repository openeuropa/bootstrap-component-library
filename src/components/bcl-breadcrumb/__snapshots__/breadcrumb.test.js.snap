// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Breadcrumb renders correctly 1`] = `
<jest>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      class="breadcrumb"
    >
      <li
        class="breadcrumb-item d-none d-md-block"
      >
        <a
          href="/example.html"
        >
          Home
        </a>
      </li>
      <li
        class="breadcrumb-item d-none d-md-block"
      >
        <a
          href="/example.html"
        >
          European Union
        </a>
      </li>
      <li
        class="breadcrumb-item"
      >
        <a
          href="/example.html"
        >
          <svg
            class="d-md-none ms-0 me-2 me-2-5 bi icon--s"
          >
            <use
              xlink:href="/icons.svg#arrow-left"
            />
          </svg>
          About the European Union
        </a>
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item d-none d-md-block active"
      >
        News
      </li>
    </ol>
  </nav>
</jest>
`;
